### PR TITLE
Prepare for release 2.17.2 

### PR DESCRIPTION
**Pull request by j-rivero**
# 🎈 Release

Preparation for 2.17.2 release.

[Comparison to <x.y.z>: https://github.com/gazebosim/<REPO>/compare/<LATEST_TAG_BRANCH>...<RELEASE_BRANCH>](https://github.com/gazebosim/gz-cmake/compare/ignition-cmake2_2.17.1...ign-cmake2)

## Checklist
- [x] Asked team if this is a good time for a release
- [ ] There are no changes to be ported from the previous major version
- [ ] No PRs targeted at this major version are close to getting in
- [ ] Bumped minor for new features, patch for bug fixes
- [x] Updated changelog
- [ ] Updated migration guide (as needed)
- [ ] Link to PR updating dependency versions in appropriate repository in [gazebo-release](https://github.com/gazebo-release) (as needed): <LINK>

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 ## Gazebo CMake 2.x
 
+### Gazebo CMake 2.17.2 (2024-04-07)
```
Comment (iche033) [inline Changelog.md:3]:
```suggestion
### Gazebo CMake 2.17.2 (2024-05-07)
```

File: Changelog.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
+1. Backport #402: Replace `exec_program` with `execute_process`
+    * [Pull request #402: Replace `exec_program` with `execute_process` (#427](https://github.com/gazebosim/gz-cmake/pull/402: Replace `exec_program` with `execute_process` (#427)
```
Comment (iche033) [inline Changelog.md:6]:
```suggestion
    * [Pull request #402](https://github.com/gazebosim/gz-cmake/pull/402)
```

File: Changelog.md (hunk 1 above)
Comment (j-rivero) [reply Changelog.md:3]: :face_with_spiral_eyes:

**Comment by scpeters** [comment]
for reference: https://github.com/gazebosim/gz-cmake/compare/ignition-cmake2_2.17.1...ign-cmake2

**Pull request merged**